### PR TITLE
SUAVE dns

### PR DESCRIPTION
**Pull request by jinmel**
## 📝 Summary

Introduces SuaveEthRemoteBackendEnpointV2 flag to support multi domain kettle. `suave.eth.remote-endpoint` now accepts rpc endpoint configuration formatted in `chain_id=rpc_url`

## 📚 References
Generalizing suave execution namespace for multiple chains:  https://github.com/flashbots/suave-specs/issues/79

---

* [x] I have seen and agree to CONTRIBUTING.md

### CODE REVIEW

**Comment by jinmel** [comment]
Should I keep the backwards compatibility by introducing new precompiles with different signatures?  (i.e. buildEthBlockV2, simulateBundleV2 ...) It seems like the integration tests are failing due to changes in precompiles

**Review by ferranbt — CHANGES_REQUESTED** [review]
Overall looks good, could you write a unit test for the `contracts_suave.go` change and then another e2e integration? (`suave/e2e`).

File: cmd/geth/forgecmd.go [hunk 1]
```diff
@@ -36,6 +36,10 @@ var (
 		Name:  "whitelist",
 		Usage: `The whitelist external endpoints to call`,
 	}
+	dnsRegistryForgeFlag = &cli.StringSliceFlag{
```
Comment (ferranbt) [inline cmd/geth/forgecmd.go:39]: Love it!

**Comment by jinmel** [comment]
@ferranbt Yeah I was also wondering if the kettle should set the port as well but that wouldn't be dns anymore. And also, if we are thinking of 
```solidity
Suavex.simulateTxn("goerlisuavex", ...)
```
we should call this endpoint aliasing rather than dns.

**Comment by jinmel** [comment]
@ferranbt After giving it some thought, I believe it's a better practice for developers if we resolve an alias name directly to its full URL, port included. It simplifies things by hiding the complexities of the URL endpoint. Let me know if you think otherwise

File: core/vm/contracts_suave.go [hunk 2]
```diff
@@ -220,12 +220,20 @@ func (s *suaveRuntime) doHTTPRequest(request types.HttpRequest) ([]byte, error)
...
-	for _, allowedDomain := range s.suaveContext.Backend.ExternalWhitelist {
-		if allowedDomain == "*" || allowedDomain == parsedURL.Hostname() {
-			allowed = true
-			break
+	// resolve dns if possible
+	if endpoint, ok := s.suaveContext.Backend.DnsRegistry[request.Url]; ok {
+		request.Url = endpoint
+		allowed = true
+	} else {
+		// check if the domain is allowed
```
Comment (ferranbt) [inline core/vm/contracts_suave.go:228]:
```suggestion
parsedURL, err := url.Parse(request.Url)
if err != nil {
     panic(err)
}
```
You do not have to parse the url if its a dns domain.

File: core/vm/contracts_suave.go [hunk 3]
```diff
@@ -220,12 +220,20 @@ func (s *suaveRuntime) doHTTPRequest(request types.HttpRequest) ([]byte, error)
 	}
 
 	var allowed bool
-	for _, allowedDomain := range s.suaveContext.Backend.ExternalWhitelist {
-		if allowedDomain == "*" || allowedDomain == parsedURL.Hostname() {
-			allowed = true
-			break
+	// resolve dns if possible
```
Comment (ferranbt) [inline core/vm/contracts_suave.go:223]: Line 217, do you have to parse the url? I guess that is only required if not a dns entry.

**Review by ferranbt — APPROVED** [review]
Only one comment to resolve but it is minor since the tests work.

File: core/vm/contracts_suave.go (hunk 3 above)
Comment (jinmel) [reply core/vm/contracts_suave.go:223]: good point!

**Pull request merged**